photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,282 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Module which provides classes to perform PSF Photometry"""
+
+from __future__ import division
+import abc, six
+import numpy as np
+from astropy.table import Table, vstack
+from photutils.psf import subtract_psf
+
+
+__all__ = ['PSFPhotometryBase', 'NStarPSFPhotometry']
+
+
+class PSFPhotometryBase(object):
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def do_photometry(self):
+        pass
+
+
+class NStarPSFPhotometry(PSFPhotometryBase):
+    """
+    This is an implementation of the NSTAR algorithm proposed by Stetson
+    (1987) to perform point spread function photometry in crowded fields.
+
+    This class basically implements the loop FIND, GROUP, NSTAR,
+    SUBTRACT, FIND until no more stars are detected.
+    """
+
+    def __init__(self, find, group, bkg, psf, fitter, niters, fitshape):
+        """
+        Attributes
+        ----------
+        find : an instance of any StarFinderBase subclasses
+        group : an instance of any GroupStarsBase subclasses
+        bkg : an instance of any BackgroundBase2D (?) subclasses
+        psf : Fittable2DModel instance
+        fitter : Fitter instance
+        niters : int
+            number of iterations for the loop FIND, GROUP, SUBTRACT, NSTAR
+        fitshape : array-like
+            rectangular shape around the center of a star which will be used
+            to collect the data to do the fitting
+        """
+
+        self.find = find
+        self.group = group
+        self.bkg = bkg
+        self.psf = psf
+        self.fitter = fitter
+        self.niters = niters
+        self.fitshape = fitshape
+
+        @property
+        def niters(self):
+            return self._niters
+        @niters.setter
+        def niters(self, niters):
+            if isinstance(niters, int) and niters > 0:
+                self._niters = niters
+            else:
+                raise ValueError('niters is not defined properly, '
+                                 'received niters = {}'.format(niters))
+        @property
+        def fitshape(self):
+            return self._fitshape
+        @fitshape.setter
+        def fitshape(self, fitshape):
+            fitshape = np.asarray(fitshape)
+            if len(fitshape) == 2 and np.all(fitshape) > 0:
+                self._fitshape = fitshape
+            else:
+                raise ValueError('fitshape is not defined properly, '
+                                 'received fitshape = {}'.format(fitshape))
+
+    def __call__(self, image):
+        """
+        Parameters
+        ----------
+        image : array-like, ImageHDU, HDUList
+            image to perform photometry
```

Comment:
Similar to what I said three lines below
```
`~astropy.io.fits.ImageHDU`
```
will generate a link to the ImageHDU docs. (the `~` means " display on the the last part, the IMAGEHDU and not the astropy.io.fits when you render the link in html)